File: BookingService/BookingWeb/src/app/reservation/contracts/FlightFilter.ts
export class FlightFilter {
    date: string = "";
    departureLocation: string = "";
    destinationLocation: string = "";
    numberOfTickets: number = 0;
  
    constructor(
      date: string,
      departureLocation: string,
      destinationLocation: string,
      numberOfTickets: number
    ) {
      this.date = date;
      this.departureLocation = departureLocation;
      this.destinationLocation = destinationLocation;
      this.numberOfTickets = numberOfTickets;
    }
  }
  

File: BookingService/APIGateway/presentation/contracts/flightFilter.go
package contracts

import (
	"bytes"
	"encoding/json"
	"io"
)

type FlightFilter struct {
	Date                string
	DepartureLocation   string
	DestinationLocation string
	NumberOfTickets     int
}

func (f *FlightFilter) FromJSON(r io.Reader) error {
	e := json.NewDecoder(r)
	return e.Decode(f)
}

func (f *FlightFilter) ToJSON() ([]byte, error) {
	return json.Marshal(f)
}

func (f *FlightFilter) ToReader() (io.Reader, error) {
	jsonBytes, err := f.ToJSON()
	if err != nil {
		return nil, err
	}
	return bytes.NewReader(jsonBytes), nil
}
